odxtools/diagcomm.py

```diff
@@ -0,0 +1,241 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional, cast
+from xml.etree import ElementTree
+
+from .admindata import AdminData
+from .audience import Audience
+from .createsdgs import create_sdgs_from_et
+from .element import IdentifiableElement
+from .exceptions import odxraise, odxrequire
+from .functionalclass import FunctionalClass
+from .nameditemlist import NamedItemList
+from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId, OdxLinkRef
+from .odxtypes import odxstr_to_bool
+from .specialdatagroup import SpecialDataGroup
+from .state import State
+from .statetransition import StateTransition
+from .utils import dataclass_fields_asdict
+
+if TYPE_CHECKING:
+    from .diaglayer import DiagLayer
+
+
+class DiagClassType(Enum):
+    STARTCOMM = "STARTCOMM"
+    STOPCOMM = "STOPCOMM"
+    VARIANTIDENTIFICATION = "VARIANTIDENTIFICATION"
+    READ_DYN_DEFINED_MESSAGE = "READ-DYN-DEFINED-MESSAGE"
+    DYN_DEF_MESSAGE = "DYN-DEF-MESSAGE"
+    CLEAR_DYN_DEF_MESSAGE = "CLEAR-DYN-DEF-MESSAGE"
```

Comment:
from spec?

Reply:
yes. This class was just moved over from `singleecujob.py` (`DiagService` would have needed it, too, but it did not internalize the `DIAGNOSTIC-CLASS` sub-element...)